### ISSUE TITLE
saidam017 - SetToken that use Aave leverage module prone to DoS, attacker can send collateral tokens with LTV = 0

### ISSUE THREAD

**Issue by sherlock-admin** [sherlock-audit/2023-05-Index-judging#90]
saidam017

high

# SetToken that use Aave leverage module prone to DoS, attacker can send collateral tokens with LTV = 0

## Summary

When an aToken with LTV = 0 is owned by setToken, it will be restricted by Aave to perform operations that require not having collateral with 0 LTV. Some of the operations are crucial and required for `AaveV3LeverageModule` and can break `AaveV3LeverageModule` functionalities if attacker send this aToken to setToken. 

## Vulnerability Detail

Aave restrict and will revert some operations that have `hasZeroLtvCollateral` is true. such as withdraw, transfer. 

https://github.com/aave/aave-v3-core/blob/master/contracts/protocol/libraries/logic/ValidationLogic.sol#L606-L609

```solidity
    require(
      !hasZeroLtvCollateral || reserve.configuration.getLtv() == 0,
      Errors.LTV_VALIDATION_FAILED
    );
```

If these functions revert, it will break some setToken functionality : 

1. delever function that rely on withdraw functionality : 
https://github.com/sherlock-audit/2023-05-Index/blob/main/index-protocol/contracts/protocol/modules/v1/AaveV3LeverageModule.sol#L313-L359

2. redeem function on basic issuance module : 
https://github.com/sherlock-audit/2023-05-Index/blob/main/index-protocol/contracts/protocol/modules/v1/BasicIssuanceModule.sol#L129-L163

3. redeem function on debt issuance module : 
https://github.com/sherlock-audit/2023-05-Index/blob/main/index-protocol/contracts/protocol/modules/v1/DebtIssuanceModuleV2.sol#L142-L190

Attacker can simply send LTV = 0 aToken to setToken so make this happen.

## Impact

This will break at least the mentioned functionalities on mentioned modules. 

## Code Snippet

https://github.com/aave/aave-v3-core/blob/master/contracts/protocol/libraries/logic/ValidationLogic.sol#L606-L609
https://github.com/sherlock-audit/2023-05-Index/blob/main/index-protocol/contracts/protocol/modules/v1/AaveV3LeverageModule.sol#L313-L359
https://github.com/sherlock-audit/2023-05-Index/blob/main/index-protocol/contracts/protocol/modules/v1/DebtIssuanceModuleV2.sol#L142-L190

## Tool used

Manual Review

## Recommendation

Implement functionality to withdraw, transfer or set that asset with LTV = 0 as non-collateral on Aave leverage module and Aave leverage strategy extension

**Comment by said017** [comment]
Escalate for 10 USDC

I think this is valid duplicate of #https://github.com/sherlock-audit/2023-05-Index-judging/issues/159

**Comment by said017** [comment]
I dont understand what are you talking about, I literally  mentioned in my issue `perform operations that require not having collateral with 0 LTV`. You comment in my issue that actually put Morpho report on the context of this index protocol. You should escalate other report that literally copy and paste the Morpho report sir

**Comment by Evert0x** [comment]
@0xffff11 what is the reason this was removed as a duplicate? This issue is copied from the report but doesn't mention the poison attack.

**Comment by said017** [comment]
Can elaborate more on which poison attack that isn't mentioned? it does mention the functions that will be DoS-ed by this particular attack (withdraw and redeem).

I would argue that while #159 explain the impact (the revert functions), but doesn't specifically mentioned how collateral with LTV come to the set Token in the first place. But that doesn't mean his issue is invalid, because the risk is real as it is completed by the sponsor in the comment how it happened.

**Comment by hildingr** [comment]
> @0xffff11 what is the reason this was removed as a duplicate? This issue is copied from the report but doesn't mention the poison attack.

"Attacker can simply send LTV = 0 aToken to setToken so make this happen." 

This is the "poison attack" which is not relevant to index protocol. I argue that this is invalid since it points to a false issue and does not find the actual issue. The issue confirmed by sponsor [here](https://github.com/sherlock-audit/2023-05-Index-judging/issues/159#issuecomment-1600066030) is the issue explained in detail in #315 and not identified in this issue.

**Comment by said017** [comment]
> "Attacker can simply send LTV = 0 aToken to setToken so make this happen."

While I agree that my statement is incorrect with this one, I would argue that this mean your issue and the main issue basically user mistake/error and self rekt since , " Because the strategy extension is designed to work with only one aToken anyway. "

And better you use escalate feature next time sir.

**Comment by hildingr** [comment]
> > "Attacker can simply send LTV = 0 aToken to setToken so make this happen."
> 
> While i agree that my statement is incorrect with this one, I would argue that this mean your issue and the main issue basically user mistake/error and self rekt since , " Because the strategy extension is designed to work with only one aToken anyway. "

Go ahead and read his reply again. He explicitly says it is VALID but that it will not be a problem in the way they will use the protocol for now. Here is what they will do  "keep the logic as is and list this as an explicit limitation. ".

**Comment by said017** [comment]
Yes technically valid but not expected to be used that way, isn't that user mistake? the problem can happened if user put two component (not expected to be used this way), and one of them have 0 LTV (self rekt if he put the 0 LTV active as collateral himself, if Aave set the LTV to 0 later after it is activated as collateral, i have seen some are reported similar case and categorized as external pause and invalidated).

Regardless, I agree that I mistaken the scenario by sending LTV = 0 Token, and this is invalid.

And after check again, If two components case is the main argument, isn't the LTV = 0 can be easily removed as collateral anyway? 

https://github.com/sherlock-audit/2023-05-Index/blob/3190057afd3085143a31746d65045a0d1bacc78c/index-protocol/contracts/protocol/modules/v1/AaveLeverageModule.sol#L594-L606

https://github.com/sherlock-audit/2023-05-Index/blob/3190057afd3085143a31746d65045a0d1bacc78c/index-protocol/contracts/protocol/modules/v1/AaveLeverageModule.sol#L1006-L1073

But I would argue the main and yours either fall into user error or external pause, and can mitigated by internal functionality, but let the judge decide that.

**Comment by hildingr** [comment]
A configuration of the Set protocol that was perfectly fine in AAVE2 can lead to a serious issue when the protocol uses AAVE3 instead, seems like a valid to me, the judge and sponsor. 

It is a general purpose basket building protocol that supports multiple modules. Index or somebody else could configure such a SetToken in the future. That is why they will be adding this limitation to their documentation. Why would they do that if it is not relevant?

**Comment by said017** [comment]
Your submission is plain invalid anyway " Delevering can be blocked by other positions".

The blocking can easily removed by this function :

https://github.com/sherlock-audit/2023-05-Index/blob/3190057afd3085143a31746d65045a0d1bacc78c/index-protocol/contracts/protocol/modules/v1/AaveLeverageModule.sol#L594-L606

**Comment by hildingr** [comment]
No that is not how the index protocol works. Please review the permission flow of the index protocol. BaseManager is the manager of the AaveLeverageModule, AaveLeverageStrategyExtension is the adapter of the BaseManager that can invoke calls to the aave module through interactManager(). The manager of the aaveleverageStrategyextension can NOT invoke a call to that function because that functionality does not exist in the extension. 

Index has purposely built a protocol with a higher level of decentralization where the manager is constrained by the functionality of the adapter.

**Comment by said017** [comment]
"No that is not how the index protocol works." Now you talking not how it works, this is technically can be removed and technically VALID, BaseManager can just transfer the manager role to EOA and do this call.

Again, that is just plain invalid. How can the manager can insert set tokens collaterals in the first place? Obviously via initial EOA Manager? so what is stopping BaseManager to transfer ownership to EOA? 

I Guess no more debate, your are correct pointing out mistake in my issue, but your issue also plain invalid.

**Comment by hildingr** [comment]
So you just agreed that they have not accounted for this issue. They first have to identify that it is blocked and why, then they have to transfer the manager to an EOA (might require DAO vote) which means that no adapter can control the SetToken during that time and then they have to remove and reconfigure the manager. What are the risks of doing this?  Until this has been done deleveraging is blocked. 

Also: If the SetToken is locked, then this will not be possible at all in which case delevering is impossible until unlocked.

**Comment by said017** [comment]
>  transfer the manager to EOA (might require DAO vote) 

This is just pure speculation.

Function is there in the code, flow path to execute are expected (EOA <-> BaseManager), transferManager is in the code, so they accounted emergency transfer manager.

**Comment by hildingr** [comment]
> > EOA (might require DAO vote)
> 
> This is just pure speculation.
> 
> Function is there in the code, flow path to execute are expected, transferManager is in the code, so they accounted emergency transfer manager.

That is why I said "might", not relevant to the argument. I think we have both made our cases here. Thank you for a good discussion.

**Comment by 0xffff11** [comment]
The main reason this issue was not included in the first results is because as @hildingr  described, the report does not describe a clear way on how to do it and it has incorrect statements. It showed me that watson did not fully understand the issue and he just copied a previous issue that is not completely valid for index

**Comment by hrishibhat** [comment]
Result:
Low
Unique
Based on the above discussion this is not a valid duplicate of #159 & #315 
https://github.com/sherlock-audit/2023-05-Index-judging/issues/303#issuecomment-1651199528

**Comment by sherlock-admin2** [comment]
Escalations have been resolved successfully!

Escalation status:
- [said017](https://github.com/sherlock-audit/2023-05-Index-judging/issues/90/#issuecomment-1610041886): rejected
